src/ui/actions/timeline.ts

```diff
@@ -605,6 +619,59 @@ export function setFocusRegionStartTime(startTime: number, sync: boolean): UIThu
   };
 }
 
+const shouldRerunAnalysisForBreakpoint = (
+  state: UIState,
+  bp: Breakpoint,
+  focusRegion: FocusRegion
+): boolean => {
+  // Copy-pasted-tweaked comment from actions/messages.ts:
+  // Soft Focus: The frontend only needs to refetch data if:
+  // 1. The most recent time it requested data "overflowed" (too many hits to send them all), or
+  // 2. The new focus region is outside of the most recent region we ran breakpoint analysis for.
+  //
+  // There are two things to note about the second bullet point above:
+  // 1. When devtools is first opened, there is no focused region.
+  //    This is equivalent to focusing on the entire timeline, so we often won't need to refetch messages when focusing for the first time.
```

Comment:
This is actually not quite the same as console messages. When we first fetch console messages we call `findMessages` (not `findMessagesInRange`) which will attempt to find console messages throughout the recording *regardless of what is attempting to load*. With something like messages we can do this because it's static data that gets pulled out by basic data retrieval. However, by default I believe we attempt to load something like the last minute of the recording. So an initial analysis will run for that portion, not the entire recording, and we need to record that. Another way of saying this is: we care not about the focus region, but about the intersection of the focus region and the loading regions.